patches/tModLoader/Terraria/ModLoader/Default/ModAccessorySlotPlayer.cs

```diff
@@ -193,16 +285,242 @@ public override void SendClientChanges(ModPlayer clientPlayer)
+		}
+		else {
+			return Loader.IsAccessoryInConflict(Player, exAccessorySlot[functional], functional, Terraria.UI.ItemSlot.Context.ModdedAccessorySlot);
+		}
+	}
+
+	// Extended Loadout, contains Item instances for ModAccessorySlot for a specific loadout index.
+	/// <summary>
+	/// <see cref="ExEquipmentLoadout"/> holds loadout items for loadouts not in use. <see cref="ModAccessorySlot"/> corollary to <see cref="EquipmentLoadout"/>.
+	/// Note that when a loadout is selected, the items in <see cref="ExEquipmentLoadout"/> are swapped into <see cref="exAccessorySlot"/>, etc. and the loadout is left with empty Item instances. The active loadout items are stored on the player and the <see cref="ExEquipmentLoadout"/> is left empty, this is the same approach used for vanilla loadouts as well (<see cref="Player.Loadouts"/>).
+	/// </summary>
+	internal sealed class ExEquipmentLoadout
+	{
+		private readonly string identifier;
+
+		public ExEquipmentLoadout(int loadoutIndex, int slotCount, EquipmentLoadout loadoutReference)
+		{
+			this.LoadoutIndex = loadoutIndex;
+			this.LoadoutReference = loadoutReference;
+			this.identifier = $"loadout_{loadoutIndex}";
+
+			this.ResetAndSizeAccessoryArrays(slotCount);
+		}
+
+		public int LoadoutIndex { get; }
+
+		public EquipmentLoadout LoadoutReference { get; }
+
+		/// <inheritdoc cref="ModAccessorySlotPlayer.exAccessorySlot"/>
+		public Item[] ExAccessorySlot { get; private set; } = [];
+
+		/// <inheritdoc cref="ModAccessorySlotPlayer.exDyesAccessory"/>
+		public Item[] ExDyesAccessory { get; private set; } = [];
+
+		/// <inheritdoc cref="ModAccessorySlotPlayer.exHideAccessory"/>
+		public bool[] ExHideAccessory { get; private set; } = [];
+
+		public void SaveData(TagCompound tag)
+		{
+			tag[identifier] = new TagCompound {
+				["items"] = ExAccessorySlot.Select(ItemIO.Save).ToList(),
+				["dyes"] = ExDyesAccessory.Select(ItemIO.Save).ToList(),
+				["hidden"] = ExHideAccessory.ToList(),
+			};
+		}
+
+		/// <summary>
+		/// Loads data for this loadout and updates this instance accordingly.
+		/// Returns a collection of <see cref="SlotInfo"/> objects for slots, which are not added to the loadout,
+		/// because <see cref="ModAccessorySlot.HasEquipmentLoadoutSupport"/> changed since the last save.
+		/// </summary>
+		/// <param name="tag">The <see cref="TagCompound"/> from which to load the data</param>
+		/// <param name="order">Saved slot names in order.</param>
+		/// <param name="slots">Slot name to slot info mapping.</param>
+		/// <returns>
+		/// A collection of <see cref="SlotInfo"/> objects for slots, which are not added to the loadout,
+		/// because <see cref="ModAccessorySlot.HasEquipmentLoadoutSupport"/> changed since the last save.
+		/// </returns>
+		public IReadOnlyList<SlotInfo> LoadData(
+			TagCompound tag,
+			List<string> order,
+			Dictionary<string, (int SlotType, bool HasLoadoutSupport)> slots)
+		{
+			List<SlotInfo> result = [];
```

Comment:
This should be `List<Item> itemsToDrop`. Modify the return type and `extraItems` accordingly